lib/layout/dummyLayoutV2.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * dummyLayoutV2.js
+ *
+ * this creates a basic poorly laid out graph
+ *
+ * @author John O'Brien.
+ */
+
+// Create a graph instance.
+
+function buildGraph(graph) {
+  var lib = './../';
+  var Graph = require(lib + 'model/graph');
+  var Node = require(lib + 'model/node');
+  var Port = require(lib + 'model/port');
+  var CardinalPortPosition = require(lib + 'model/cardinalPortPosition');
+  var nodes = graph.getNodes();
+  //loop for building the nodes
+  for(var i = 0; i < nodes.length; i++){
+    nodes[i].setPosition(i * 2, i * 2);
+	
+    var ins = nodes[i].getInputs();
+    var outs = nodes[i].getOutputs();
+	
+    for(var iterator = 0; iterator < ins.length; iterator++){
+      ins[ iterator ].setPosition(CardinalPortPosition('E', 0.5));
```

Comment:
Make sure to use the "new" keyword with CardinalPortPosition here. If you don't, "this" inside of the newly instantiated object will not be bound to the newly created instance. IIRC, it will be bound to whatever the current context is (in this case, the buildGraph function?) which will result in weird errors that aren't easy to trace.